src/Message/Payload/Builder/MessagePayloadBuilder.php

```diff
@@ -92,11 +92,15 @@ public function withClaim($claim, $claimValue = null): MessagePayloadBuilderInte
         return $this;
     }
 
-    public function withMessagePayloadClaims(MessagePayloadInterface $payload): MessagePayloadBuilderInterface
-    {
+    public function withMessagePayloadClaims(
+        MessagePayloadInterface $payload,
+        array $exclusions = []
+    ): MessagePayloadBuilderInterface {
         /** @var Claim $claim */
         foreach ($payload->getToken()->getClaims() as $claim) {
-            $this->builder->withClaim($claim->getName(), $claim->getValue());
+            if (!in_array($claim->getName(), $exclusions)) {
```

Comment:
This doesn't really feel like a responsibility of the payload builder.

External services are still in charge of figuring out what are those "exclusions", why not filter the payload there in the first place?

Reply:
they do, but providing the list of exclusions when calling the builder